### ISSUE TITLE
kftray assumes that portforwarding is still active after a system restart

### ISSUE THREAD

**Issue by caoimhebyrne** [hcavarsan/kftray#324]
**Describe the bug**
If you start port-forwarding via kftray, restart your computer, and open it again, the services you port-forwarded will still appear as active, but if you attempt to connect to one of these services, it will fail.

![2024-10-16 at 12 22 37](https://github.com/user-attachments/assets/ff8d264d-67ab-49ea-be09-0f935e909a58)

**To Reproduce**
1. Open kftray
2. Enable port-forwarding for one or more services & wait for them to start
3. Restart your computer
4. Re-open kftray
5. Look at the services previously port-forwarded, they are still appearing as active

**Expected behavior**
After an application/system restart, it should either:
1. Re-start the port-forwarding for the previous services
2. Not show the services as active

**Screenshots**
N/A?

**Desktop (please complete the following information):**
 - OS: macOS Sequoia 15.0.1 (24A348)
 - Version: kftray 0.14.8

**Additional context**
If you press one of the toggle switches for a service, it will eventually show "No portforwarding process found", and pressing "Stop All" just takes a weirdly long amount of time to complete.

![2024-10-16 at 12 28 20](https://github.com/user-attachments/assets/d2a6e064-73cc-4fb9-9afb-9340f98c585a)

No additional context, I have collected logs, but I'm unsure if I can share them without leaking sensitive information, there's a lot in them, so it would be time-consuming to sanitize.

**Comment by caoimhebyrne** [comment]
It's worth mentioning that closing kftray before restarting is not actually a requirement here, I just tested by opening -> port forwarding -> restarting -> opening and I can see the same behavior.

**Comment by hcavarsan** [comment]
@caoimhebyrn, im on it. should have a new version out tomorrow with some checks when the app starts

**Comment by hcavarsan** [comment]
hey @caoimhebyrne,

[release v0.14.9](https://github.com/hcavarsan/kftray/releases/tag/v0.14.9) should fix the issue.

now, if you close the app, it'll ask you to stop or keep port-forwards running, and they’ll auto-resume next time. also made the start/stop faster.

pls try it out and let me know if it's working now. thanks!

**Comment by hcavarsan** [comment]
@caoimhebyrne  just to clarify, did this happen only after a system restart? if you manually click the close button on the app, it should show the popup asking to stop or keep port-forwards running. it’s possible the system restart sends a sigkill signal

**Comment by caoimhebyrne** [comment]
@hcavarsan I've tried both closing and reopening the app, and a system restart. It never prompted me to stop or keep running, it just keeps them running (doing so successfully).

**Comment by hcavarsan** [comment]
it's weird cuz i tested on different OSs and the dialog showed up correctly... anyway, i'll open another issue to look into it and close this one since the initial issue is resolved, ok?

**Comment by caoimhebyrne** [comment]
sounds good @hcavarsan

**Comment by hcavarsan** [comment]
great, thanks for the report!
